pkg/provider/azure/keyvault/keyvault.go

```diff
@@ -74,15 +75,24 @@ func newClient(ctx context.Context, store esv1alpha1.GenericStore, kube client.C
 		store:     store,
 		namespace: namespace,
 	}
-	azClient, vaultURL, err := anAzure.newAzureClient(ctx)
 
-	if err != nil {
-		return nil, err
+	clientSet, err := anAzure.setAzureClientWithServicePrincipal(ctx)
+	if clientSet {
```

Comment:
I think we can simplify this for better readability: 

```go
if ok {
  return anAzure, err
}
```

Reply:
This was what I did, but original tests start to fail, because they check if returned provider is nil if err is not nil. I'm relatively new to go, but one should not use in any way the returned value if err is not nil. So I'll modify the original tests, too.